tests/operators/test_agnostic_load_file.py

```diff
@@ -644,34 +333,71 @@ def test_load_file_using_file_connection_fails_inexistent_conn(caplog, sample_da
         "output_table": Table(table_name=OUTPUT_TABLE_NAME, **sql_server_params),
     }
     with pytest.raises(BackfillUnfinished) as exec_info:
-        test_utils.create_and_run_task(sample_dag, load_file, (), task_params)
+        with sample_dag:
+            load_file(**task_params)
+        test_utils.run_dag(sample_dag)
+
     expected_error = "Failed to execute task: The conn_id `fake_conn` isn't defined."
     assert expected_error in caplog.text
 
 
-def create_task_parameters(database_name, file_type):
-    sql_server_params = test_utils.get_default_parameters(database_name)
-
-    task_params = {
-        "path": str(pathlib.Path(CWD.parent, f"data/sample.{file_type}")),
-        "file_conn_id": "",
-        "output_table": Table(table_name=OUTPUT_TABLE_NAME, **sql_server_params),
-    }
-    return task_params
-
-
+@pytest.mark.integration
 @pytest.mark.parametrize(
     "sql_server", ["snowflake", "postgres", "bigquery", "sqlite"], indirect=True
 )
 @pytest.mark.parametrize("file_type", ["parquet", "ndjson", "json", "csv"])
-def test_load_file(sample_dag, sql_server, file_type):
+def test_load_file(sample_dag, sql_server, file_type, test_table):
     database_name, sql_hook = sql_server
 
-    task_params = create_task_parameters(database_name, file_type)
-    test_utils.create_and_run_task(sample_dag, load_file, (), task_params)
+    with sample_dag:
+        load_file(
+            path=str(pathlib.Path(CWD.parent, f"data/sample.{file_type}")),
+            file_conn_id="",
+            output_table=test_table,
+        )
+    test_utils.run_dag(sample_dag)
+    df = sql_hook.get_pandas_df(f"SELECT * FROM {test_table.qualified_name()}")
 
-    qualified_name = task_params["output_table"].qualified_name()
-    df = sql_hook.get_pandas_df(f"SELECT * FROM {qualified_name}")
+    assert len(df) == 3
+    expected = pd.DataFrame(
+        [
+            {"id": 1, "name": "First"},
+            {"id": 2, "name": "Second"},
+            {"id": 3, "name": "Third with unicode पांचाल"},
+        ]
+    )
+    df = df.rename(columns=str.lower)
+    df = df.astype({"id": "int64"})
+    expected = expected.astype({"id": "int64"})
+    assert_frame_equal(df, expected)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "sql_server",
+    [
+        "postgres",
+        "bigquery",
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "test_table",
+    [{"is_temp": False, "param": {"schema": "custom_schema"}}],
+    indirect=True,
+)
+@pytest.mark.parametrize("file_type", ["csv"])
+def test_load_file_with_named_schema(sample_dag, sql_server, file_type, test_table):
+    database_name, sql_hook = sql_server
+
+    with sample_dag:
+        load_file(
+            path=str(pathlib.Path(CWD.parent, f"data/sample.{file_type}")),
+            file_conn_id="",
+            output_table=test_table,
+        )
+    test_utils.run_dag(sample_dag)
+    df = sql_hook.get_pandas_df(f"SELECT * FROM {test_table.qualified_name()}")
```

Comment:
@utkarsharma2 we've run into issues with hanging tests in the past when people run `get_pandas_df` like this. Instead, could you please create an @adf decorated function that validates the test? You should see an example of this in the `test_transform` file

Reply:
@dimberman, I think that would be because of conflicting table names, but since now we are using fixture to get tables, it should not reoccur. But we are adopting this as best practice I'll change it.

Reply:
I believe it is worth using the `get_pandas_df` - it keeps the tests simple enough.